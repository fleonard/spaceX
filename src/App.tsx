import React from "react";
import { BrowserRouter, Switch, Redirect, Route } from "react-router-dom";
import { Box } from "@material-ui/core";

import { GlobalStyles } from "./globalStyles";

import Dragons from "./views/Dragons/Dragons";
import Rockets from "./views/Rockets/Rockets";
import DragonPage from "./views/DetailPages/DragonPage";
import RocketPage from "./views/DetailPages/RocketPage";
import Nav from "./components/Nav/Nav";

const App: React.FC = () => {
  return (
    <BrowserRouter>
      <GlobalStyles />
      <Box component="main" className="container">
        <Nav />
        <Switch>
          <Route path="/rockets" component={Rockets} exact />
          <Route path="/rockets/:id" component={RocketPage} />
          <Route path="/dragons" component={Dragons} exact />
          <Route path="/dragons/:id" component={DragonPage} />
          <Redirect from="*" to="/rockets" />
        </Switch>
      </Box>
    </BrowserRouter>
  );
};

export default App;
