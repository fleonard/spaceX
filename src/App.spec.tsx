import React from "react";
import { render, waitFor } from "@testing-library/react";
import "jest-styled-components";

import App from "./App";

describe("Application Component", () => {
  it("renders correctly", () => {
    const { container } = render(<App />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the rocket list as default page", async () => {
    const { getByText } = render(<App />);

    await waitFor(() => expect(getByText("Falcon 1")).toBeDefined());
    await waitFor(() => expect(getByText("Falcon 9")).toBeDefined());
    await waitFor(() => expect(getByText("Falcon Heavy")).toBeDefined());
    await waitFor(() => expect(getByText("Starship")).toBeDefined());
  });
});
