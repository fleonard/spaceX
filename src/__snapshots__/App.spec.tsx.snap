// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application Component renders correctly 1`] = `
.c8 {
  display: grid;
  padding: 30px 0;
  place-items: center;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 15px 20px;
}

.c9.c9 {
  color: #000;
}

.c1 {
  cursor: pointer;
  height: 30px;
  position: relative;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: 0.5s ease-in-out;
  transition: 0.5s ease-in-out;
  width: 30px;
}

.c1 span {
  background: #fff;
  border-radius: 9px;
  display: block;
  height: 4px;
  left: 0;
  opacity: 1;
  position: absolute;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: 0.25s ease-in-out;
  transition: 0.25s ease-in-out;
  width: 100%;
}

.c1 span:first-child {
  top: 0px;
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

.c1 span:nth-child(2) {
  left: 0;
  opacity: 1;
  top: 13px;
}

.c1 span:nth-child(3) {
  top: 25px;
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

.c4 {
  background: #000;
  color: #fff;
  height: calc(100vh - 60px);
  padding: 20px;
  position: absolute;
  left: 0;
  top: 60px;
  -webkit-transform: translateX(-100%) translateZ(0);
  -ms-transform: translateX(-100%) translateZ(0);
  transform: translateX(-100%) translateZ(0);
  -webkit-transform-origin: 0% 0%;
  -ms-transform-origin: 0% 0%;
  transform-origin: 0% 0%;
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
  width: 270px;
  z-index: 10;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0;
  text-align: right;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 10px;
  width: 100%;
}

.c6 a {
  border-bottom: 1pt solid #252525;
  color: #fff;
  padding: 10px 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  width: 100%;
}

.c0 {
  background-color: #000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 15px 20px;
  position: relative;
}

.c0.sticky {
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 10;
}

.c0 svg {
  display: block;
  fill: #fff;
  height: auto;
  width: 200px;
}

.c2 {
  display: none;
  padding: 0;
}

.c3 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  box-sizing: content-box;
  padding: 0 10px;
}

.c3 a {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 10px 0;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
}

.c3 a::after {
  background: #fff;
  bottom: 0;
  color: transparent;
  content: ".";
  height: 1px;
  left: 0;
  margin: auto;
  position: absolute;
  right: 0;
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
  width: 0%;
}

.c3 a:hover:after {
  width: 100%;
}

@media (min-width:768px) {
  .c1 {
    display: none;
  }
}

@media (min-width:768px) {
  .c4 {
    display: none;
  }
}

@media (min-width:768px) {
  .c2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<main
  class="MuiBox-root MuiBox-root-1 container"
>
  <div
    class="c0 false"
  >
    <svg>
      spacex-logo.svg
    </svg>
    <div
      aria-controls="mobile-menu"
      aria-expanded="false"
      aria-label="Toggle menu"
      class="c1"
    >
      <span />
      <span />
      <span />
    </div>
    <ul
      class="MuiList-root c2 MuiList-padding"
    >
      <li
        class="MuiBox-root MuiBox-root-2 c3"
      >
        <a
          aria-current="page"
          class="active"
          href="/rockets"
        >
          Rockets
        </a>
      </li>
      <li
        class="MuiBox-root MuiBox-root-3 c3"
      >
        <a
          href="/dragons"
        >
          Dragons
        </a>
      </li>
    </ul>
    <div
      aria-hidden="true"
      class="MuiBox-root MuiBox-root-4 c4"
      id="mobile-menu"
    >
      <ul
        class="MuiList-root c5 MuiList-padding"
      >
        <li
          class="MuiBox-root MuiBox-root-5 c6"
        >
          <a
            aria-current="page"
            class="active"
            href="/rockets"
          >
            Rockets
          </a>
        </li>
        <li
          class="MuiBox-root MuiBox-root-6 c6"
        >
          <a
            href="/dragons"
          >
            Dragons
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    class="c7"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4"
    >
      Rockets
    </h4>
    <div
      class="c8"
    >
      <div
        class="MuiCircularProgress-root c9 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
  </div>
</main>
`;
